controllers/mover/rclone/builder.go

```diff
@@ -33,21 +34,38 @@ import (
 // data mover
 const defaultRcloneContainerImage = "quay.io/backube/volsync-mover-rclone:latest"
 
-// rcloneContainerImage is the container image name of the rclone data mover
-var rcloneContainerImage string
+// Command line flag will be checked first
+// If command line flag not set, the RELATED_IMAGE_ env var will be used
+const rcloneContainerImageFlag = "rclone-container-image"
+const rcloneContainerImageEnvVar = "RELATED_IMAGE_RCLONE_CONTAINER"
 
 type Builder struct{}
 
 var _ mover.Builder = &Builder{}
 
-func Register() {
-	flag.StringVar(&rcloneContainerImage, "rclone-container-image",
-		defaultRcloneContainerImage, "The container image for the rclone data mover")
+func init() {
+	// Set default rclone container image - will be used if both command line flag and env var are not set
+	viper.SetDefault(rcloneContainerImageFlag, defaultRcloneContainerImage)
+}
```

Comment:
Can this just be done in the Register() function instead of having an init?

Reply:
@JohnStrunk I've made another commit to re-arrange a bit (including the change you suggested above) and allow unit tests to set their own viper & flagset to hopefully avoid issues with parallel tests.  Please let me know what you think, it does end up a little more convoluted, but hopefully helps with tests.  It still may be possible to get errors with the env var tests, although I haven't encountered any yet.

Reply:
What about something like having `RegisterWithFlags()`?
so we get:
```golang
func Register() error {
    return RegisterWithFlags(flag.CommandLine, viper.GetViper())
}

// This could even be little "registerWithFlags" since it's only being accessed by tests
func RegisterWithFlags(flags, viper) error {
    b := &Builder {
        viper: viper,
        flags: flags
    }
    // initFlags contents
    mover.Register(b)
}
```

I think this could get rid of the getViper/getFlagset functions and just let everything directly reference b.viper, etc. For example:
```golang
func (rb *Builder) getRcloneContainerImage() string {
	return rb.viper.GetString(rcloneContainerImageFlag)
}
```

Does this seem less convoluted?

Reply:
Hey John, thanks for the suggestion, I do like this idea, but one question:  One thing I was trying to avoid was calling register multiple times, resulting in the overall mover.Catalog getting the same builder appended to it multiple times by different tests.  Generally most of the mover tests don't need register (we have one specific Register() test which is good), but I ended up initing the flags again for those tests to force the mover image to be initialized.

What do you think about me updating initFlags (can rename to newBuilder or something of the sort) to work like your registerWithFlags() suggestion but just without the mover.Register() call?

i.e. 
```
func Register() error {
  b := newBuilder(flag.CommandLine, viper.GetViper())
  mover.Register(b)
}

func newBuilder(flags, viper) error {
      b := &Builder {
        viper: viper,
        flags: flags
    }
   b.viper.SetDefault(rsyncContainerImageFlag, defaultRsyncContainerImage)
   ... etc
}
```

Reply:
That works. :+1:

Reply:
Thanks!  I've updated the PR again with this suggestion